src/msw/headerctrl.cpp

```diff
@@ -874,7 +1000,167 @@ bool wxHeaderCtrl::MSWOnNotify(int idCtrl, WXLPARAM lParam, WXLPARAM *result)
         return true;
     }
 
-    return wxHeaderCtrlBase::MSWOnNotify(idCtrl, lParam, result);
+    return wxControl::MSWOnNotify(idCtrl, lParam, result);
+}
+
+// ============================================================================
+// wxHeaderCtrl implementation
+// ============================================================================
+
+// ----------------------------------------------------------------------------
+// wxHeaderCtrl construction/destruction
+// ----------------------------------------------------------------------------
+
+void wxHeaderCtrl::Init()
+{
+    m_nativeControl = NULL;
+}
+
+bool wxHeaderCtrl::Create(wxWindow *parent,
+                          wxWindowID id,
+                          const wxPoint& pos,
+                          const wxSize& size,
+                          long style,
+                          const wxString& name)
+{
+    long newStyle = style | wxCLIP_CHILDREN | wxTAB_TRAVERSAL;
+    if ( !wxWindow::Create(parent, id, pos, size, newStyle, name) )
+        return false;
+
+    m_nativeControl = new wxMSWHeaderCtrl(*this);
+    if ( !m_nativeControl->Create(this,
+                                  wxID_ANY,
+                                  wxDefaultPosition,
+                                  wxDefaultSize,
+                                  ApplyHeaderReorderFlagToStyle(wxNO_BORDER),
+                                  wxMSWHeaderCtrlNameStr) )
+        return false;
+
+    Bind(wxEVT_SIZE, &wxHeaderCtrl::OnSize, this);
+
+    return true;
+}
+
+wxHeaderCtrl::~wxHeaderCtrl()
+{
+    delete m_nativeControl;
```

Comment:
Is this really needed? Normally window children should be destroyed automatically when the window itself is.

If it is needed, could you please add a comment explaining why?